gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFsHelper.java

```diff
@@ -98,37 +52,13 @@ public void close() throws FileBasedHelperException {
   }
 
   @Override
-  public List<String> ls(String path) throws FileBasedHelperException {
-    List<String> results = new ArrayList<String>();
-    try {
-      lsr(new Path(path), results);
-    } catch (IOException e) {
-      throw new FileBasedHelperException("Cannot do ls on path " + path + " due to " + e.getMessage(), e);
-    }
-    return results;
-  }
-
-  public void lsr(Path p, List<String> results) throws IOException {
-    if (!this.fs.getFileStatus(p).isDir()) {
-      results.add(p.toString());
-    }
-    for (FileStatus status : this.fs.listStatus(p)) {
-      if (status.isDir()) {
-        lsr(status.getPath(), results);
-      } else {
-        results.add(status.getPath().toString());
-      }
-    }
-  }
-
-  @Override
   public InputStream getFileStream(String path) throws FileBasedHelperException {
     try {
       Path p = new Path(path);
-      InputStream in = this.fs.open(p);
+      InputStream in = this.getFileSystem().open(p);
```

Comment:
Please put this in `try (InputStream in = this.getFileSystem().open(new Path(path)))`.